docs/site/Loopback-component-authentication.md

```diff
@@ -0,0 +1,631 @@
+**returning** the authentication strategy which has that specific **name** and
+has been `registered` as an **extension** of the aforementioned **extension
+point**.
+
+```ts
+@extensionPoint(
+  AuthenticationBindings.AUTHENTICATION_STRATEGY_EXTENSION_POINT_NAME,
+  {scope: BindingScope.TRANSIENT},
+) //this needs to be transient, e.g. for request level context.
+export class AuthenticationStrategyProvider
+  implements Provider<AuthenticationStrategy | undefined> {
+  constructor(
+    @extensions()
+    private authenticationStrategies: Getter<AuthenticationStrategy[]>,
+    @inject(AuthenticationBindings.METADATA)
+    private metadata?: AuthenticationMetadata,
+  ) {}
+  async value(): Promise<AuthenticationStrategy | undefined> {
+    if (!this.metadata) {
+      return undefined;
+    }
+    const name = this.metadata.strategy;
+    const strategy = await this.findAuthenticationStrategy(name);
+    if (!strategy) {
+      // important to throw a non-protocol-specific error here
+      let error = new Error(`The strategy '${name}' is not available.`);
+      Object.assign(error, {
+        code: AUTHENTICATION_STRATEGY_NOT_FOUND,
+      });
+      throw error;
+    }
+    return strategy;
+  }
+
+  async findAuthenticationStrategy(name: string) {
+    const strategies = await this.authenticationStrategies();
+    const matchingAuthStrategy = strategies.find(a => a.name === name);
+    return matchingAuthStrategy;
+  }
+}
+```
+
+In order for your custom authentication strategy to be **found**, it needs to be
+**registered**.
+
+**Registering** a custom authentication strategy `BasicAuthenticationStrategy`
+as an extension of the
+`AuthenticationBindings.AUTHENTICATION_STRATEGY_EXTENSION_POINT_NAME` extension
+point in an application `application.ts` is as simple as:
+
+```ts
+registerAuthenticationStrategy(this, BasicAuthenticationStrategy);
+```
+
+## Summary
+
+We've gone through the main steps for adding `authentication` to your LoopBack 4
+application.
+
+Your `application.ts` should look similar to this:
+
+```ts
+export class MyApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(RestApplication)),
+) {
+  constructor(options?: ApplicationConfig) {
+    super(options);
+
+    /* set up miscellaneous bindings */
+
+    //...
+
+    // load the authentication component
+    this.component(AuthenticationComponent);
+
+    // register your custom authentication strategy
+    registerAuthenticationStrategy(this, BasicAuthenticationStrategy);
+
+    // use your custom authenticating sequence
+    this.sequence(MyAuthenticatingSequence);
+
+    this.static('/', path.join(__dirname, '../public'));
+
+    this.projectRoot = __dirname;
+
+    this.bootOptions = {
+      controllers: {
+        dirs: ['controllers'],
+        extensions: ['.controller.js'],
+        nested: true,
+      },
+    };
+  }
+```
+
+You can find a working **example** of a LoopBack 4 shopping cart application
```

Comment:
nitpick: I'm not quite sure if we need to bold the "example" word.. similar to a few incidents towards the end of the last section.  It could be just personal preference.